codegen/object.go

```diff
@@ -212,28 +212,37 @@ func (f *Field) doWriteJson(val string, remainingMods []string, astType *ast.Typ
 		}
 		var arr = "arr" + strconv.Itoa(depth)
 		var index = "idx" + strconv.Itoa(depth)
+		var wg = "wg" + strconv.Itoa(depth)
 		var usePtr bool
 		if len(remainingMods) == 1 && !isPtr {
 			usePtr = true
 		}
 
-		return tpl(`{{.arr}} := graphql.Array{}
+		return tpl(`
+			{{.arr}} := make(graphql.Array, len({{.val}}))
+			var {{.wg}} sync.WaitGroup
+			{{.wg}}.Add(len({{.val}}))
 			for {{.index}} := range {{.val}} {
-				{{- if not .isScalar }}
-					rctx := &graphql.ResolverContext{
-						Index: &{{.index}},
-						Result: {{ if .usePtr }}&{{end}}{{.val}}[{{.index}}],
-					}
-					ctx := graphql.WithResolverContext(ctx, rctx)
-				{{- end}}
-				{{.arr}} = append({{.arr}}, func() graphql.Marshaler {
-					{{ .next }} 
-				}())
+				go func({{.index}} int) {
```

Comment:
we should probably avoid creating a new gofunc if:
 - there is only one element in the array
 - the child field doesn't have a resolver (accessing properties is cheap and should happen sequentially)